app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java

```diff
@@ -430,13 +392,79 @@ public void clearSections() {
       zimFile.setData(uri);
       startActivity(zimFile);
     }
+  }
 
-    wasHideToolbar = isHideToolbar;
-    booksAdapter = new BooksAdapter(books, this);
+  private void setupDocumentParser() {
+    documentParser = new DocumentParser(new DocumentParser.SectionsListener() {
+      @Override
+      public void sectionsLoaded(String title, List<DocumentSection> sections) {
+        for (DocumentSection section : sections) {
+          if (section.title.contains("REPLACE_")) {
+            section.title = getResourceString(getApplicationContext(), section.title);
+          }
+        }
+        documentSections.addAll(sections);
+        if (title.contains("REPLACE_")) {
+          tableDrawerAdapter.setTitle(getResourceString(getApplicationContext(), title));
+        } else {
+          tableDrawerAdapter.setTitle(title);
+        }
+        tableDrawerAdapter.setSections(documentSections);
+        tableDrawerAdapter.notifyDataSetChanged();
+      }
 
-    searchFiles();
-    tabRecyclerView.setAdapter(tabsAdapter);
-    new ItemTouchHelper(tabCallback).attachToRecyclerView(tabRecyclerView);
+      @Override
+      public void clearSections() {
+        documentSections.clear();
+        tableDrawerAdapter.notifyDataSetChanged();
+      }
+    });
+  }
+
+  private void setTabListener() {
+    tabsAdapter.setTabClickListener(new TabsAdapter.TabClickListener() {
+      @Override
+      public void onSelectTab(View view, int position) {
+        hideTabSwitcher();
+        selectTab(position);
+
+        /* Bug Fix
+         * Issue #592 in which the navigational history of the previously open tab (WebView) was
+         * carried forward to the newly selected/opened tab; causing erroneous enabling of
+         * navigational buttons.
+         */
+        updateBottomToolbarArrowsAlpha();
+      }
+
+      @Override
+      public void onCloseTab(View view, int position) {
+        closeTab(position);
+      }
+    });
+  }
+
+  private void setTableDrawerInfo() {
+    tableDrawerRight.setLayoutManager(new LinearLayoutManager(this));
+    tableDrawerAdapter = setupTableDrawerAdapter();
+    tableDrawerRight.setAdapter(tableDrawerAdapter);
+    tableDrawerAdapter.notifyDataSetChanged();
+  }
+
+  private void setupTabsAdapter() {
+    tabsAdapter = new TabsAdapter(this, webViewList);
+    tabsAdapter.registerAdapterDataObserver(new RecyclerView.AdapterDataObserver() {
+      @Override
+      public void onChanged() {
+        updateTabSwitcherIcon();
+      }
+    });
+  }
+
+  private void addFileReader() {
+    FileReader fileReader;
+    fileReader = new FileReader();
+    documentParserJs = fileReader.readFile("js/documentParser.js", this);
```

Comment:
I think you can reduce the above 3 lines to this `documentParserJs = new FileReader() .readFile("js/documentParser.js", this);`

Reply:
@abdulwd Done.